src/compiler/transformers/declarations.ts

```diff
@@ -1137,7 +1137,7 @@ namespace ts {
                         fakespace.symbol = props[0].parent!;
                         const declarations = mapDefined(props, p => {
                             if (!isPropertyAccessExpression(p.valueDeclaration)) {
-                                return undefined;
+                                return undefined; // TODO: Handle element access expressions that created late bound names (rather than silently omitting them)

```

Comment:
I'd rather have a bug tracking this or an implementation in the PR. Preferably the first.

Reply:
Well, the TODO exists so we can discuss which we'd prefer~

Reply:
(Which do we prefer?)

Reply:
Opened #33569 and amended the comment to reference the issue.